### PR TITLE
Enable specification of cache location, maintaining backwards compatibility

### PR DESCRIPTION
**Pull request by austindyoung**
This PR addresses issue #58. It is a sparser version of PR #70 that maintains backwards with the current dev branch at this time.

### CODE REVIEW

File: lib/UserAgentApplication.ts [hunk 1]
```diff
@@ -183,24 +183,28 @@ namespace Msal {
...
             }
-
-            this.redirectUri = window.location.href.split("?")[0].split("#")[0];
+            this.redirectUri = window.location.href.split('?')[0].split('#')[0];
             this.postLogoutredirectUri = this.redirectUri;
             this._loginInProgress = false;
             this._acquireTokenInProgress = false;
             this._renewStates = [];
             this._activeRenewals = {};
-            this._cacheStorage = new Storage(this._cacheLocation); //cache keys msal
```
Comment (rohitnarula7176) [inline lib/UserAgentApplication.ts]: @austindyoung  Why did you delete this line. We need a reference to this._cacheStorage in our code.

File: lib/UserAgentApplication.ts [hunk 2]
```diff
@@ -183,24 +183,28 @@ namespace Msal {
...
+            tokenReceivedCallback: tokenReceivedCallback,
+            validateAuthority?: boolean,
+            cacheLocation = 'sessionStorage'
+        ) {
+            this.cacheLocation = cacheLocation
             this.clientId = clientId;
-
             this.validateAuthority = validateAuthority === true;
-            this.authority = authority ? authority : "https://login.microsoftonline.com/common";
+            this.authority = authority || 'https://login.microsoftonline.com/common';
```
Comment (rohitnarula7176) [inline lib/UserAgentApplication.ts]: The condition you removed does the same thing. Any particular reason for this change.

File: lib/UserAgentApplication.ts (hunk 1 above)
Comment (austindyoung) [inline lib/UserAgentApplication.ts]: @rohitnarula7176 
On line 193, the pre-existing setter for cache location is used. It was implemented but not used.

In the current code, this._cacheLocation gets defaulted to 'sessionStorage', so there was no way to set the cache location on the UserAgentApplication instance without mutating it.

File: lib/UserAgentApplication.ts (hunk 2 above)
Comment (austindyoung) [inline lib/UserAgentApplication.ts]: @rohitnarula7176 This is a style change; Two instances of `authority` vs one.

**Comment by rohitnarula7176** [comment]
@austindyoung thank you for the pull request. The issue is fixed in dev.
